src/graphql/schemas/resolvers/orderResolver.ts

```diff
@@ -48,17 +52,55 @@ class OrderResolver {
     try {
       const res = await this.supabaseService.getOrders(args);
 
-      const { data, error, count } = res;
+      const { data: orders, error, count } = res;
 
       if (error) {
-        console.warn(
-          `[ContractResolver::orders] Error fetching orders: `,
-          error,
-        );
-        return { data };
+        console.warn(`[OrderResolver::orders] Error fetching orders: `, error);
+        return { orders };
       }
 
-      return { data, count: count ? count : data?.length };
+      const groupedOrders = _.groupBy(orders, (order) => order.chainId);
+
+      const ordersAfterCheckingValidity = await Promise.all(
+        Object.entries(groupedOrders).map(async ([chainId, ordersForChain]) => {
+          const chainIdParsed = parseInt(chainId);
+          const hypercertExchangeClient = new HypercertExchangeClient(
+            chainIdParsed,
+            // @ts-expect-error - wagmi and viem have different typing
+            new ethers.JsonRpcProvider(getRpcUrl(chainIdParsed)),
+          );
+
+          const validityResults =
+            await hypercertExchangeClient.checkOrdersValidity(
```

Comment:
Shouldn't this be for all? E.g. a currency get's added, balance of fraction got updated, order becomes valid because of starttime, etc..

Reply:
Can you clarify? This runs the checkOrderValidity function on the contract, which should include all of these errors right?

Or do you mean it should check all orders for that tokenID? In that case, that happens inside the validate function.

Reply:
This filter only checks when they aren't currently invalid, or am I misunderstanding the code?

` ordersForChain.filter((order) => !order.invalidated),`

Reply:
Yes, it currently only checks for orders that have not been invalidated yet, which means that if you have an order that would be revalidated, that wouldn't happen. Would you expect that to? Can remove the invalidated check, downside being everything gets checked everytime it gets fetched.

Reply:
ok, let's leave this for now. We discussed previously that if we put it in the API, we might want to add a cron job that does the full validaty check. It'll probably need to only check based on invalid codes that are not permanent. Let's not shoehorn this in rn but put it in an issue

Reply:
https://github.com/hypercerts-org/hypercerts-api/issues/117